units/src/fee.rs

```diff
@@ -146,19 +149,24 @@ impl FeeRate {
     /// fee is a non-integer amount, the amount is rounded up, ensuring that the transaction fee is
     /// enough instead of falling short if rounded down.
     ///
-    /// Returns [`None`] if overflow occurred.
+    /// Returns [`NumOpResult::Error`] if overflow occurred.
     #[must_use]
-    pub const fn checked_mul_by_weight(self, weight: Weight) -> Option<Amount> {
-        // No `?` operator in const context.
+    pub const fn checked_mul_by_weight(self, weight: Weight) -> NumOpResult<Amount> {
         match self.to_sat_per_kwu().checked_mul(weight.to_wu()) {
-            Some(mul_res) => match mul_res.checked_add(999) {
-                Some(add_res) => match Amount::from_sat(add_res / 1000) {
-                    Ok(fee) => Some(fee),
-                    Err(_) => None,
-                },
-                None => None,
+            Some(fee) => match Amount::from_sat(fee) {
+                Ok(amt) => match amt.to_sat().checked_add(999) {
+                    Some(add_op) => match add_op.checked_div(1000) {
```

Comment:
No need for `checked_div`, 1000 is not 0.

Reply:
Good point.  I thought about that too but mistakenly thought if add_op was zero it would matter.  You're right though that this doesn't need to be checked.